Message.js

```diff
@@ -1,18 +1,68 @@
 class Message {
-  constructor(message) {
+  constructor(affirmations, mantras) {
+    this.affirmations = affirmations;
+    this.mantras = mantras;
     this.message = message;
+    this.selectedMessageList = null;
     this.renderedAffirmations = [];
     this.renderedMantras = [];
+    this.resetMessage = null;
   }
 
-  populateAffirmations() {
-    // this.renderedAffirmations.push(this.message);
+  generateMessage() {
+    if (this.selectedMessageList === 'affirmations') {
+      this.message = this.affirmations[this.getRandomIndex(this.affirmations)];
+    } else if (this.selectedMessageList === 'mantras') {
+      this.message = this.mantras[this.getRandomIndex(this.mantras)];
+    }
+  }
+
+  getRandomIndex(inputMessageList) {
+    return Math.floor(Math.random() * inputMessageList.length);
+  }
+
+  populateRenderedAffirmationMantra() {
+    if (this.selectedMessageList === 'affirmations') {
+      this.renderedAffirmations.push(this.message);
+    } else if (this.selectedMessageList === 'mantras') {
+        this.renderedMantras.push(this.message);
+    }
+  }
+
+  removeCurrentMessageFromAffirmationMantra() {
+    if (this.selectedMessageList === 'affirmations') {
+      var index = this.affirmations.indexOf(this.message);
+      this.affirmations.splice(index, 1); //WHY DOES THIS ALSO SPLICE AFFIRMATIONS ARRAY?
+    } else if (this.selectedMessageList === 'mantras') {
+      var index = this.mantras.indexOf(this.message);
+      this.mantras.splice(index, 1); //WHY DOES THIS ALSO SPLICE MANTRAS ARRAY?
+    }
+    console.log(this.affirmations.length, this.mantras.length);
+  }  
 
-    this.renderedAffirmations = this.renderedAffirmations + this.message;
+  resetMessageAffirmations() {
+    this.resetMessage = null; //only need to reset once...
+    if (this.affirmations.length === 0) {
```

Comment:
You might be able to shorten this to a falsey value since you're checking if there's nothing in the array

Reply:
good catch... will take a look!